deploy-site.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+echo -e "\033[0;32mDeploying website...\033[0m"
+
+cd site && npx hexo generate
+
+# delete old deploy branch
+git branch -D deploy
+
+git checkout -b deploy
+
+# Add changes to git.
+git add -f public
+
+# Commit changes.
+msg="chore(*): adding public folder `date`"
+if [ $# -eq 1 ]
+  then msg="$1"
+fi
+git commit -m "$msg"
+
+cd ../
+
+git subtree split -P site/public -b deploy-dist
+
+# Push source and build repos.
+git push -f embark-site deploy-dist:gh-pages
```

Comment:
**IMPORTANT** the origin is called `embark-site`, so everyone who wants to deploy needs to make sure to have it set up accordingly

Reply:
Can we use a git command to check whether the `embark-site` remote is setup in the local repository?

Can we employ some logic similar to that of `scripts/release.js` re: `--release-branch`, pertaining to branches of `embark-framework/embark`? One, check that the current branch is a match for `--release-branch`, which defaults to `master` if not specified in the cli. Two, check that the local and remote branches are in sync.

If we make it so this script is invoked by `scripts/release.js` then it's not an issue. But I'm thinking that's probably not always the case. For example, fixing typos or grammatical errors, or adding a post — such changes won't always coincide with an embark release. So it seems we may be invoking this script manually sometimes and automatically other times.

Reply:
> One, check that the current branch is a match for --release-branch, which defaults to master if not specified in the cli.

Since the release branch will always be `gh-pages` there's probably no point in checking if the current branch is a match for sth. like `--release-branch`. So even if the default is `master`, the website will be pushed to `gh-pages` on the embark-site origin. Unless we have a reason to make this configurable, but I don't see any at this point.

> Two, check that the local and remote branches are in sync.

First I thought: Yes we can do that. But then I realized, what if I really just want to put out some typo fixes and have to do that from a different branch (maybe prior to a feature that has landed incl. docs but should be released yet). In such case, there is no remote branch that can be in sync with the local one.

Reply:
> If we make it so this script is invoked by scripts/release.js then it's not an issue. But I'm thinking that's probably not always the case. For example, fixing typos or grammatical errors, or adding a post — such changes won't always coincide with an embark release. So it seems we may be invoking this script manually sometimes and automatically other times.

Exactly. We definitely want to keep this separate. Embark and Website should be releasable without releasing the other.

Reply:
> Since the release branch will always be `gh-pages` there's probably no point...

I meant with respect to local branch and the corresponding remote branch on github `embark-framework/embark`, not `embark-framework/embark-site`.

If I'm locally on `master` then before deploying to remote `gh-pages` branch on `embark-framework/embark-site` is it important to check that remote `master` branch on `embark-framework/embark` matches local `master`?

Reply:
> But then I realized, what if I really just want to put out some typo fixes... there is no remote branch that can be in sync with the local one.

Shouldn't the typo fixes land in `master` of `embark-framework/embark` before the deployment to `gh-pages` of `embark-framework/embark-site`?

Reply:
Absolutely. But imagine this scenario (which is very close to what we've discussed with regards to our release flow):

- Typo fixes landed in master
- New features landed in master in the meantime, including their docs
- typo fixes need to be deployed, but we don't wanna put the feature docs out yet because Embark hasn't made a new release yet and is not planning to do so right now

^ For this I would probably just go back in time, check out that commit that has the typo fixes, or even make a branch and cherry-pick several fixes if needed and run a deploy from there.

This is pretty much the same thing we've done for `4.0.x` releases for Embark, just with the crucial difference that we don't keep the commit history for the released website.

Reply:
> This is pretty much the same thing we've done for `4.0.x` releases for Embark, just with the crucial difference that we don't keep the commit history for the released website.

Thanks, the explanation makes sense.

> typo fixes need to be deployed, but we don't wanna put the feature docs out yet because Embark hasn't made a new release yet and is not planning to do so right now

Would it make sense if deployed as nightlies of the website, corresponding to nightlies of embark?